### PR TITLE
Change: Overhaul Power Action Logic

### PR DESCRIPTION
**Pull request by martinmckenna**
## Description

Tech-debt Tuesday item

Completely overhauls power action logic. This was one of the most annoying parts of the codebase to work with imo.

## Type of Change
- Non breaking change ('update', 'change')

## Applicable E2E Tests

finding which ones broke is a major TODO....

## Note to Reviewers

Some highlights and enhancements include:

1. Adds loading and error state for _all_ power actions
   * these now live inside the dialog, rather than in a toast. yay
2. Rebooting now asks what config you want to boot (enhancement)
   * the one exception is the _Reboot_ button on the CardView, but this is not a regression, so I'm okay with fixing that in another PR.
   * ^^^ to fix that, we're going to need to either pre-load configs or load the configs when LinodesLanding mounts.
3. The Config Drawer Selection lives inside the Boot Action Confirmation Dialog component
   * This seems a little confusing, but those components go hand-in-hand, so I argue it makes the developer experience better.
4. The Delete Dialog was separated out from the Boot Action Confirmation Dialog on the Linodes Landing.
    * in other words, they don't share state anymore because they're two completely different dialogs that do different things.
6. Powering on now has a dialog to confirm. This wasn't necessarily a required change, but we didn't have great UX for loading and error states here prior. I think this is definitely something we should iterate on.
5. LOTS of prop drilling.

### CODE REVIEW

File: src/features/linodes/LinodesLanding/powerActions.ts [hunk 1]
```diff
@@ -1,116 +1,121 @@
-import * as moment from 'moment';
+/**
+ * legacy implementation. Keeping around until we know why this
+ * was generating phony events.
+ */
```
Comment (martinmckenna) [inline src/features/linodes/LinodesLanding/powerActions.ts]: take note of the comment here. Maybe @rjstires would care to comment?

Comment (rjstires) [reply src/features/linodes/LinodesLanding/powerActions.ts]: We in the technical realm, @martinmckenna, call them "simulated" events.

The faked event were a means to an end. The API events responses were taking so long that from the time the user initiated the action to the time the progress actually started was significant. So we threw in some simulated events.

Between 300+ ms request/response cycle for the reboot, and the 800ms+ events endpoint response time we easily broke over 1000ms just waiting for that first event to kick off the reboot. And that's supposing that the events table actually updated to in_progress or scheduled when you sent your first request 1000s later.

You very well might see a regression here depending on the speed of the network and API.

Godspeed young Marty.

Comment (martinmckenna) [reply src/features/linodes/LinodesLanding/powerActions.ts]: 🙏

Comment (martinmckenna) [reply src/features/linodes/LinodesLanding/powerActions.ts]: wait hang on can't you make that same argument for all events that have some sort of progress on them?

Comment (rjstires) [reply src/features/linodes/LinodesLanding/powerActions.ts]: Yes. I believe this particular instance came up as part as a review (you can imagine...). I think we did the same with the Linode power control component or whatever we called it.

Comment (martinmckenna) [reply src/features/linodes/LinodesLanding/powerActions.ts]: ok. in this case, I'm fine with the regression

Comment (rjstires) [reply src/features/linodes/LinodesLanding/powerActions.ts]: Me too.

File: src/features/linodes/LinodesLanding/DeleteDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
```
Comment (Jskobos) [inline src/features/linodes/LinodesLanding/DeleteDialog.tsx]: I made a separate deletion dialog bc the work you're doing here hadn't been done and PowerActions didn't have the right error handling. Now that this refactor is done, we should probably open a ticket to fold this in to `PowerActionsDialogOrDrawer`.

File: src/features/linodes/LinodesLanding/DeleteDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  const [isDeleting, setDeleting] = React.useState<boolean>(false);
+  const [errors, setErrors] = React.useState<
+    Linode.ApiFieldError[] | undefined
+  >(undefined);
+
+  React.useEffect(() => {
+    if (props.open) {
+      /**
+       * reset error state, username, and disabled status when we open the modal
+       * intentionally not resetting comments
```
Comment (Jskobos) [inline src/features/linodes/LinodesLanding/DeleteDialog.tsx]: Comments seem copy-pasted

File: src/features/linodes/LinodesLanding/DeleteDialog.tsx (hunk 2 above)
Comment (martinmckenna) [reply src/features/linodes/LinodesLanding/DeleteDialog.tsx]: I believe this component should live on its own since it's not really a power action.

**Comment by rjstires** [comment]
🎉
